### ISSUE TITLE
OpenID support (or SAML)

### ISSUE THREAD

**Issue by olivierlambert** [louislam/uptime-kuma#553]
That would be cool to get login working via an OpenID provider (using SSO in short, with Keycloak for example).

I know that `keycloak-js` can do it (https://www.keycloak.org/securing-apps/vue is a complete walk-through), or even https://www.npmjs.com/package/vue-oidc-client as a more generic one.

I'm not fluent enough in Vue, but I can do the tests for you on a dedicated branch if you like :+1:

**Comment by eldiaboloz** [comment]
Not exactly what you want, but if you put the service behind [oauth2-proxy](https://github.com/oauth2-proxy/oauth2-proxy) and then disable auth in uptime-kuma, you can integrate with keycloak and all other oauth2 providers that oauth2-proxy supports.
This will become a problem if there is support for more than one user in the future, but for now works great.

**Comment by olivierlambert** [comment]
Thanks! That's interesting but I'd like to have the status page public, and be able to login for the dashboard (as now, but with Oauth).

Is that doable with oauth2-proxy?

**Comment by eldiaboloz** [comment]
Yes it is possible - the authetication is done in keycloak only.  You need to set only which group from keycloak has access 
`OAUTH2_PROXY_KEYCLOAK_GROUPS: /some-group`
For the public status page you need to allow some urls:
`OAUTH2_PROXY_SKIP_AUTH_REGEX=(/status|/assets/.*|/icon.svg)`
[Here](https://oauth2-proxy.github.io/oauth2-proxy/docs/configuration/oauth_provider/#keycloak-auth-provider) is more documentation on using keycloak provider in oauth2-proxy
There is a little difference if you use env variables ( via docker ) or [cfg file](https://github.com/oauth2-proxy/oauth2-proxy/blob/master/contrib/oauth2-proxy.cfg.example#L67)

**Comment by oussjarrousse** [comment]
I think adding SSO integration (OAuth or SAML) would make this APP much more interesting for small tech-teams and small organisations.

**Comment by Maven35** [comment]
+1 i am looking for SSO integration at least either OIDC or SAML since I run keycloak in my environment

**Comment by Daemonslayer2048** [comment]
If it matters at all, I also would find (basic) SAML or OIDC support useful

**Comment by xgaia** [comment]
I'm interested to the openID support to, and I'm interested to implement the functionality. @louislam , do you accept PR for this ?

**Comment by maxulm** [comment]
> FYI @eldiaboloz it works with this solution for now. Even if I'd like "real" OpenID integration, this is enough for my use case 👍

could you please post the relevant part of your docker-compose-file?
I always get a 404 of the oauth2-proxy.

**Comment by olivierlambert** [comment]
I'm not using Docker at all.

**Comment by maxulm** [comment]
> I'm not using Docker at all.
Thanks!
Maybe @eldiaboloz has a hint?

**Comment by eldiaboloz** [comment]
@maxulm 
[Here](https://gist.github.com/eldiaboloz/39c6890ceee404299144778b8729e8ee) is my working example with cleaned up "secrets" and my domain name.

I think i ended using cfg instead of .env because the skip_auth_routes needs to be an array and the regex was getting too complex to read

**Comment by maxulm** [comment]
> @maxulm [Here](https://gist.github.com/eldiaboloz/39c6890ceee404299144778b8729e8ee) is my working example with cleaned up "secrets" and my domain name.
> 
> I think i ended using cfg instead of .env because the skip_auth_routes needs to be an array and the regex was getting too complex to read

Thanks a lot!

**Comment by marekful** [comment]
Hey @louislam how you going,

How would you receive a PR adding a basic OIDC login mechanism? (Similar to [this](https://github.com/NginxProxyManager/nginx-proxy-manager/pull/2630) one.)

Just a minimal integration whereby following a successful authentication with an Identity Provider, Uptime Kuma would check if there is an existing user in its system with a username matching the one extracted from the ID token issued by the IdP and log the user in to that account. This could be done using [openid-client](https://www.npmjs.com/package/openid-client) in under 500 lines of code in a day or two.

FYI @olivierlambert

**Comment by Lanhild** [comment]
Uptime Kuma being a really great FOSS status service watcher (emphasizing on the "FOSS" as most status page providers make you pay ridiculous fees for the service it is), OIDC support would be greatly appreciated

**Comment by cwchristerw** [comment]
Related to #21

**Comment by CommanderStorm** [comment]
> Related to #2280

Not related to #2280, but an offshoot of https://github.com/louislam/uptime-kuma/issues/128:
Said issue being the one which goes into detail what systems need to be changed how to allow multiple users. PRs doing part of the work are appreciated.

The change suggested by @marekful seems quite daunting and really hard to implement+review.
Any PR in this field has to be:
- small enough to review easily (otherwise stuff slips past, which is bad in auth)
- include tests that verify the behaviour

**Comment by CrimsonFez** [comment]
A simple implementation with oidc-client-ts could achieve the same result as using ouath2-proxy. I don't think user management would have to be a thing, just check if the user has a configured group or role. All user management happens in the oidc provider.

**Comment by VladoPortos** [comment]
Yea the implementation of OIDC would be huge bonus for this app. Same as many we run keycloak to access everything and I would be super happy if we got just verification, yes this use is in this group he can log in... does not have to support full user management in app. and Log out button to work ( people tend to forgot it :D )

**Comment by sniff122** [comment]
Been thinking about deploying this at work to provide some monitoring for our developers of services and having oauth2 would be SO much easier to handle. Something like an allowed domains thing too so then we can restrict which email domains can log in

**Comment by PurseChicken** [comment]
+1000

**Comment by arunoruto** [comment]
I started using [beszel](https://beszel.dev/) and it offloads its user management to [PocketBase](https://pocketbase.io/), which enables OIDC out of the box with some popular provides predefined!
I am not really familiar with it, but it also supports databases that users can look into via a WebUI.
It would introduce an additional dependency, but offloading such things should also make it easier to maintain the core features of the project, instead of venturing into side projects.

**Comment by EHRETic** [comment]
I can only upvote Oauth/OIDC! 😊

**Comment by jannis6023** [comment]
A regex of all protected routes would be nice. Then the solution of using oauth2-proxy would be perfect. But as we want our status page to be visible to anyone, we would only need to secure /dashboard and all restricted api routes!

**Comment by thmo** [comment]
[Authentik](https://goauthentik.io/) has a guide on how to integrate with Uptime Kuma [here](https://docs.goauthentik.io/integrations/services/uptime-kuma/).

The guide has a list of paths that do _not_ need to be protected.

One merely cosmetic (I hope!) issue I noticed while testing this, is that a status page shows the knobs for editing the status page, or going to the dashboard, when one was logged in before - because there exists a `token:autoLogin` item in the browser's Local Storage. As far as I can see, this does not really allow to modify the status page, though, because the save knob does not work.

**Comment by mike12806** [comment]
Will uptime-kuma 2.0.0 include OIDC support?

**Comment by CommanderStorm** [comment]
Negative, as user management requires multi-user support.
- #128 


> [!TIP]
> Feel free to test said PR though:
> - https://github.com/louislam/uptime-kuma/pull/3571
> 
> ```bash
> npx kuma-pr M1CK431:add_basic_multiple_admin_users

**Comment by learncoder4848** [comment]
@CommanderStorm I had the implemented the working code for OIDC , can you guide me the right steps to raise this feature request PR ?

**Comment by CommanderStorm** [comment]
You can open the PR, sure. But I will only look at it after we have basic multi user support.
Before this, OIDC does not make sense imo (happy to be proven wrong)

**Comment by MikeW1901** [comment]
> You can open the PR, sure. But I will only look at it after we have basic multi user support. Before this, OIDC does not make sense imo (happy to be proven wrong)

I disagree that this doesn't have usefulness in its own right, for example being able to give access via e.g. Google Auth to a trusted set of individuals within an organisation, whilst giving one less password to worry about rotating when one of these users leaves the org.

**Comment by pro-sumer** [comment]
> Before this, OIDC does not make sense imo (happy to be proven wrong)

(Native) OIDC support would allow us to log in using a passkey (via [Pocket ID](https://pocket-id.org)) instead of username, password and TOTP 2FA code.

(But yes, we can probably already achieve that using [caddy-security](https://github.com/greenpau/caddy-security) and the like)

**Comment by learncoder4848** [comment]
> [@CommanderStorm](https://github.com/CommanderStorm) I had the implemented the working code for OIDC , can you guide me the right steps to raise this feature request PR ?

As promised much awaited feature is here 
https://github.com/louislam/uptime-kuma/pull/6232

Requesting you all to kindly take the pull, test the functionality, review it and let's push it to add this as feature.

FYI, the same is working with enterprise level SSO provider in production in my org 😎
